thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/ThriftCallServiceBuilder.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.thrift;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.concurrent.Executors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * A fluent builder to build an instance of {@link ThriftCallService}.
+ *
+ * @see ThriftCallService
+ */
+public final class ThriftCallServiceBuilder {
+    private final ImmutableMap.Builder<String, Iterable<?>> implementations = ImmutableMap.builder();
+    private boolean useBlockingTaskExecutor;
+
+    ThriftCallServiceBuilder() {}
+
+    /**
+     * Adds multiple implementations for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder implementations(Map<String, ? extends Iterable<?>> implementations) {
+        requireNonNull(implementations, "implementations");
+        if (implementations.isEmpty()) {
+            throw new IllegalArgumentException("empty implementations");
+        }
+        this.implementations.putAll(implementations);
+        return this;
+    }
+
+    /**
+     * Adds an implementation for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder addService(Object implementation) {
+        requireNonNull(implementation, "implementation");
+        return addService("", implementation);
+    }
+
+    /**
+     * Adds an implementation with key for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder addService(String key, Object implementation) {
+        requireNonNull(implementation, "implementation");
+        this.implementations.put(key, ImmutableList.of(implementation));
```

Comment:
This will overwrite the existing map entry under the specified `key`, making the following code work incorrectly:

```
builder.addService("foobar", fooService);
builder.addService("foobar", barService);
```

There are two options to fix this here:
- Use `ImmutableMultiMap.Builder` from Guava
- Check if there's already a mapping and then merge if so.


Also, it'd be nice if a user is allowed to specify multiple implementations in a single call:

```
builder.addService("foobar", fooService, barService);
// and 
builder.addService("foobar", ImmutableList.of(fooService, barService));
```

We could replace `addService()` with this new method, then.

Reply:
@trustin 
Is it okay to use like this? I also added test [here](https://github.com/line/armeria/pull/5619/files#diff-ea16fbab04daf518e353c882cc5ddf068117675e9d17026d13159e40b8886ba5).

https://github.com/line/armeria/blob/649bf67555a6d442655dce70680e7e15a499b235/thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/ThriftCallServiceBuilder.java#L34-L48